src/vm/mod.rs

```diff
@@ -352,6 +352,21 @@ fn eval_all(
     })
 }
 
+/* Run provided program in an environment specified by the user. This version of the function
+  gives the most leeway to the caller to create a custom environment.
+*/
+pub fn execute_program_with_context(
```

Comment:
If you just needed this for testing, can you instead revert the changes to this file and put the function into your test module?

Reply:
1)  we would then have to expose "eval_all"
2) we can use this in other tests

However, it is possible to just expose a version that is a function of (clarity_version, is_mainnet), and avoid exposing such a complicated method. What would you say about that?

Reply:
I'd support just adding an argument to `execute_against_version`